extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheProcessor.java

```diff
@@ -74,14 +75,14 @@ ValidationErrorBuildItem validateBeanDeployment(ValidationPhaseBuildItem validat
     }
 
     @BuildStep
-    @Record(STATIC_INIT)
+    @Record(RUNTIME_INIT)
```

Comment:
Thanks for your answer @mkouba! It works much better now. :)

The only change I had to do and I don't like is this one, but I don't think I have a choice here since the ManagedExecutor is initialized in `RUNTIME_INIT` generated bytecode (see [here](https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/SmallRyeContextPropagationRecorder.java#L51)).

Reply:
Oh, yes it's RUNTIME_INIT, I'm sorry. Hm, but I don't think you can reuse the `io.quarkus.deployment.builditem.ExecutorBuildItem`  - it's already produced by `io.quarkus.deployment.steps.ThreadPoolSetup.createExecutor()`. I'm surprised that it does not fail...

Reply:
It does fail indeed but my last commit should fix it. I only run a subset of Quarkus Maven modules with the previous commit, that why it didn't fail.

Reply:
I guess we'll have to accept the caches initialization at runtime now. This PR was created to fix a context issue during the cache computations.

Reply:
I think that you should produce the `ManagedExecutorBuildItem` inside the `SmallRyeContextPropagationProcessor#build()` method.

Reply:
Ok, I'll try that tonight (I have to go back to work right now).

Thanks a lot for your help with this!

Reply:
You're welcome :-).